### ISSUE TITLE
package-lock.json in gitignore?

### ISSUE THREAD

**Issue by ArcanisCz** [ProtonMail/WebClients#116]
Is there a reason why package-lock.json is not commited in git? It is meant to be in version control ( https://docs.npmjs.com/files/package-lock.json ), especially if you use version wildcards in package.json - it solves deterministic build problem.

Package-lock.json can be used as a checksum of circle CI node_modules cache better than package.json, especially when you are using not exact versions. Also, fresh install of this app without package-lock.json was 100s and with package-lock.json only 65s. Would be probably good idea to add this file to CircleCI cache too to speedup install step.

If for any reason do you still dont want package-lock.json in git, did you consider to not generating it at all? Adding `package-lock=false` to `.npmrc` file (its `--no-package-lock` argument of https://docs.npmjs.com/cli/install)

**Comment by vladimiry** [comment]
Package lock in general is supposed to keep the build in reproducible state, especially if running on CI (we would execute `npm ci` instead of `npm install`) and on developers machines. So I support @ArcanisCz, package lock needs to be committed.

I'd also suggest putting exact dependencies versions to package.json for clarity and update them running tool like https://www.npmjs.com/package/npm-check-updates (so lock file will be updated too if needed).

This bug looks like an instance of lock file absence https://github.com/ProtonMail/WebClient/issues/130

**Comment by dhoko** [comment]
Why the package-lock is not inside the project, because we removed it cf https://github.com/ProtonMail/WebClient/tree/4177584845da4c0be62052f5a3af2a988d522e59

We're a few devs working on the project and it's boring to commit the package-lock on the repo as it may change often. I'm agree it sucks if you want to generate the same bundle :/
We're trying to improve this to ensure we provide to everyone everything they need to be able to create the exact same bundle we made. 

It should be available soon ;) 

Yep we use ncu to update deps, the issue is the lock is not something consistent across envs. We work on 3 OS: GNU/Linux, MacOS or Windows.
